cpg_workflows/large_cohort/combiner.py

```diff
@@ -1,132 +1,129 @@
-                        f'to control behaviour',
-                    )
-    return [s for s in sequencing_groups if s.active]
-
-
-def check_duplicates(iterable):
+query getCohort($cohort: String!) {
+  cohorts(name: {eq: $cohort}) {
+    sequencingGroups {
+      id
+      analyses(type: {eq: "gvcf"}, status: {eq: COMPLETED}, active: {eq: true}) {
+        output
+      }
+    }
+  }
+}
+""",
+)
+
+GET_VDS_ANALYSIS_QUERY: DocumentNode = gql(
     """
-    Throws error if input list contains repeated items.
-    """
-    duplicates = [item for item, count in collections.Counter(iterable).items() if count > 1]
-    if duplicates:
-        raise ValueError(f'Found {len(duplicates)} duplicates: {duplicates}')
-    return duplicates
+query getVDSByAnalysisIds($vds_ids: [Int!]!) {
+  analyses(id: {in_: $vds_ids}) {
+    output
+  }
+}
+""",
+)
 
 
-def run(out_vds_path: Path, tmp_prefix: Path, *sequencing_group_ids) -> hl.vds.VariantDataset:
-    """
-    run VDS combiner, assuming we are on a cluster.
-    @param out_vds_path: output path for VDS
-    @param tmp_prefix: tmp path for intermediate fields
-    @param sequencing_group_ids: optional list of sequencing groups to subset from get_multicohort()
-    @return: VDS object
-    """
-    if can_reuse(out_vds_path):
-        return hl.vds.read_vds(str(out_vds_path))
-
-    sequencing_groups = get_multicohort().get_sequencing_groups()
-    if sequencing_group_ids:
-        sequencing_groups = [s for s in sequencing_groups if s in sequencing_group_ids]
-
-    sequencing_groups = _check_gvcfs(sequencing_groups)
-    seq_types = set(s.sequencing_type for s in sequencing_groups)
-    if len(seq_types) > 1:
-        by_seq_type = collections.defaultdict(list)
-        for s in sequencing_groups:
-            by_seq_type[s.sequencing_type].append(s.id)
-
-        raise ValueError(
-            f'Found multiple sequencing types for large cohort combiner: {by_seq_type}',
-        )
-    # else just use the first
-    sequencing_type = seq_types.pop()
-
-    params = get_config().get('large_cohort', {}).get('combiner', {})
-
-    if intervals := params.get('intervals'):
-        if isinstance(intervals, list):
-            params['intervals'] = hl.eval(
-                [hl.parse_locus_interval(interval, reference_genome=genome_build()) for interval in intervals],
-            )
-        else:
-            params['intervals'] = hl.import_locus_intervals(params['intervals'])
-    elif sequencing_type == 'exome':
-        params.setdefault('use_exome_default_intervals', True)
-    elif sequencing_type == 'genome':
-        params.setdefault('use_genome_default_intervals', True)
-    else:
-        raise ValueError(
-            'Either combiner/intervals must be set, or workflow/sequencing_type must be one of: "exome", "genome"',
-        )
+def _get_samples_from_vds(input_vds: str) -> list[str]:
+    return read_vds(input_vds).variant_data.s.collect()
```

Comment:
In an ideal world, the analysis entry in metamist which points to this VDS will contain an `analysis.sequencingGroups[.id]` attribute, collecting all the SG IDs which were part of generating the previous object.

However, it looks like the analysis results thus far haven't been registered at all ([here](https://github.com/populationgenomics/production-pipelines/blob/main/cpg_workflows/stages/large_cohort.py#L19)). You've addressed that, so it should be solved going forwards, in which case you _shouldn't_ need this check to be done on the file itself.